packages/augur-core/src/contracts/para/WethWrapperForAMMExchange.sol

```diff
@@ -137,4 +137,37 @@ contract WethWrapperForAMMExchange {
         }
         shareToken.unsafeBatchTransferFrom(_from, _to, _tokenIds, _amounts);
     }
+
+    function claimTradingProceeds(IMarket _market, address payable _shareHolder, bytes32 _fingerprint) external {
+        require(shareToken.isApprovedForAll(_shareHolder, address(this)), "Weth wrapper contract must be approved to do share token transfers on behalf of the share holder.");
+
+        uint256 _INVALID = shareToken.getTokenId(_market, 0);
```

Comment:
I'd also add a comment explaining why you're duplicating code. It's easy to miss the crucial detail and end up trying to refactor, breaking some functionality.